Sources/Syntax/Numeric.swift

```diff
@@ -0,0 +1,25 @@
+extension Syntax where A == Int, M == String {
+  /// Consumes an integer from the front of the string.
+  public static let int = Syntax<Int, String>.init(
+    monoid: .joined,
+    parse: { str in
+      let prefix = str.prefix(while: { ("0"..."9").contains($0) })
+      str.removeFirst(prefix.count)
+      return Int(String(prefix))
+  },
+    print: { "\($0)" }
+  )
+}
+
+extension Syntax where A == Double, M == String {
+  /// Consumes a double from the front of the string.
+  public static let double = Syntax<Double, String>.init(
+    monoid: .joined,
+    parse: { str in
+      let prefix = str.prefix(while: { ("0"..."9").contains($0) || $0 == "." })
```

Comment:
Maybe using `Scanner` type will be more flexible?

Reply:
Do you mean just locally here or more generally in the API?

Reply:
I meant just here, but why not both =) would be useful for all the different formats of numbers that can be there like exponential or hex (though I think it doesn't support any other systems like binary). And could use date formatter for dates in a similar manner maybe.

Reply:
Ha, we'll probably write special `Syntax` for that kind of stuff since it only performs a single peek :)

Might be nice to have some `Syntax` helpers that take various `Formatter`s, though!